cdk_eoapi/pgStacInfra.py

```diff
@@ -0,0 +1,167 @@
+        bastion_host_user_data: Union[str, aws_ec2.UserData],
+        **kwargs,
+    ) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        pgstac_db = PgStacDatabase(
+            self,
+            "pgstac-db",
+            vpc=vpc,
+            engine=aws_rds.DatabaseInstanceEngine.postgres(
+                version=aws_rds.PostgresEngineVersion.VER_14
+            ),
+            vpc_subnets=aws_ec2.SubnetSelection(
+                subnet_type=aws_ec2.SubnetType.PUBLIC
+                if public_db_subnet
+                else aws_ec2.SubnetType.PRIVATE_ISOLATED
+            ),
+            allocated_storage=db_allocated_storage,
+            instance_type=aws_ec2.InstanceType(db_instance_type),
+        )
+
+        stac_api_lambda = PgStacApiLambda(
+            self,
+            "pgstac-api",
+            api_env=dict(NAME=stac_api_lambda_name, description=f"{stage} STAC API"),
+            vpc=vpc,
+            db=pgstac_db.db,
+            db_secret=pgstac_db.pgstac_secret,
+            subnet_selection=aws_ec2.SubnetSelection(
+                subnet_type=aws_ec2.SubnetType.PUBLIC
+                if public_db_subnet
+                else aws_ec2.SubnetType.PRIVATE_WITH_EGRESS
+            ),
+        )
+
+        titiler_pgstac_api_lambda = TitilerPgstacApiLambda(
+            self,
+            "titiler-pgstac-api",
+            api_env=dict(
+                NAME=titiler_pgstac_api_lambda_name,
+                description=f"{stage} titiler pgstac API",
+            ),
+            vpc=vpc,
+            db=pgstac_db.db,
+            db_secret=pgstac_db.pgstac_secret,
+            subnet_selection=aws_ec2.SubnetSelection(
+                subnet_type=aws_ec2.SubnetType.PUBLIC
+                if public_db_subnet
+                else aws_ec2.SubnetType.PRIVATE_WITH_EGRESS
+            ),
+            buckets=titiler_buckets,
+        )
+
+        bastion_host = BastionHost(
+            self,
+            "bastion-host",
+            vpc=vpc,
+            db=pgstac_db.db,
+            ipv4_allowlist=bastion_host_allow_ip_list,
+            user_data=aws_ec2.UserData.custom(bastion_host_user_data)
+            if bastion_host_user_data
+            else aws_ec2.UserData.for_linux(),
+            create_elastic_ip=bastion_host_create_elastic_ip,
+        )
+
+        if data_access_role_arn:
+            data_access_role = aws_iam.Role.from_role_arn(
+                self,
+                "data-access-role",
+                role_arn=data_access_role_arn,
+            )
+        else:
+            data_access_role = self._create_data_access_role()
+            data_access_role = self._grant_assume_role_with_principal_pattern(
+                data_access_role, f"*{self.stack_name}*ingestor*"
+            )  # beware, there is a limit in the number of characters a role can have (64) and AWS automatically truncates the role ARN if it's too long.
+
+        stac_ingestor_env = {"REQUESTER_PAYS": "True"}
```

Comment:
why is this set to True by default?

Reply:
Interesting, if you don't include this then the HEAD request fails ? Are they requiring this to be explicitly set up so that users don't incur charges due to requester-pays bucket without knowing about it ?

Reply:
We should remember to look at how the tiler handles that